File: test_commits/python/feature_addition_changed4.py
from datetime import datetime

class Task:
    def __init__(self, name, priority="normal"):
        self.name = name
        self.completed = False
        self.created_at = datetime.now()
        self.priority = priority
        self.completed_at = None

    def mark_complete(self):
        self.completed = True
        self.completed_at = datetime.now()

class TaskManager:
    def __init__(self):
        self.tasks = []

    def add_task(self, name, priority="normal"):
        task = Task(name, priority)
        self.tasks.append(task)

    def get_all_tasks(self):
        return self.tasks

    def get_completed_tasks(self):
        return [task for task in self.tasks if task.completed]

    def get_tasks_by_priority(self, priority):
        return [task for task in self.tasks if task.priority == priority]

File: test_commits/java/feature_addition_changed4.java
import java.util.ArrayList;
import java.util.Date;
import java.util.List;

class Task {
    private String name;
    private boolean completed;
    private Date createdAt;
    private Date completedAt;
    private String priority;

    public Task(String name, String priority) {
        this.name = name;
        this.completed = false;
        this.createdAt = new Date();
        this.priority = priority;
        this.completedAt = null;
    }

    public void markComplete() {
        this.completed = true;
        this.completedAt = new Date();
    }

    public String getName() {
        return name;
    }

    public boolean isCompleted() {
        return completed;
    }

    public String getPriority() {
        return priority;
    }
}

class TaskManager {
    private List<Task> tasks = new ArrayList<>();

    public void addTask(String name, String priority) {
        tasks.add(new Task(name, priority));
    }

    public List<Task> getAllTasks() {
        return tasks;
    }

    public List<Task> getCompletedTasks() {
        List<Task> result = new ArrayList<>();
        for (Task task : tasks) {
            if (task.isCompleted()) {
                result.add(task);
            }
        }
        return result;
    }

    public List<Task> getTasksByPriority(String priority) {
        List<Task> result = new ArrayList<>();
        for (Task task : tasks) {
            if (task.getPriority().equals(priority)) {
                result.add(task);
            }
        }
        return result;
    }
}
